File: MovieApp/MovieApp/ClientApp/src/app/Movie.ts
//The Angular side models need to match the JSON data and NOT the C# Model

export class Movie {
    id: number;
    title: string;
    runTime: number;
    year: number;
    description: string;
    leadActor: string;

    constructor(id: number, title: string, runTime: number, year: number, description: string, leadActor: string) {
        this.id = id;
        this.title = title;
        this.runTime = runTime;
        this.year = year;
        this.description = description;
        this.leadActor = leadActor;
    }
}

File: MovieApp/MovieApp/Models/Movie.cs
using System;
using System.Collections.Generic;

namespace MovieApp.Models
{
    public partial class Movie
    {
        public int Id { get; set; }
        public string Title { get; set; } = null!;
        public int RunTime { get; set; }
        public int Year { get; set; }
        public string? Description { get; set; }
        public string? LeadActor { get; set; }
    }
}
